### DISCUSSION TITLE
workflows must be defined on master?

### DISCUSSION THREAD

**Discussion by mem** [community/community#25848] [Actions]
<p>Hi,</p>
<p>I’m sure there’s a bit of documentation I’m not finding…</p>
<p>Must workflows (<code>.github/workflows/my_workflow.yml</code>) be defined _and_ committed on master for them to work?</p>
<p>The case I’m thinking about here is having a slightly different workflow on master and say a release-x.y branch. If I start with a workflow on master, and later edit it on the release branch, what will get used? The version on master or the version on the release branch? The way the UI is structured suggests the answer is “the version on master”, but I cannot find a piece of documentation that says this.</p>
<p>On the same line, if I open a PR that modifes the workflow and the target branch is master, what will run? The unmodified version already on master or the modified version in the PR? If the former, how do I test that the change in the workflow is not breaking the workflow?</p>
<p>Thanks!</p>

**Comment by oldskool** [comment, marked as answer]
<p>No, a workflow can exist in any branch. It will be triggered as long as your conditions (the “on” block) matches.</p>
<p>If you open a PR with a modified workflow, all workflows that are on the “pull_request” or “push” (without target branch) condition will run based on the code from that PR/push, so with the updated workflow.</p>

**Comment by mem** [comment]
<p>Thanks! I was confused because in the UI on the screen listing the workflows that have run, there’s a menu on the right of each item that contains an option “view workflow file”, and that’s showing me the version from master.</p>
<p>Double checking, I can verify that the version that actually ran is the one from the branch.</p>

**Comment by keravt** [comment]
нет не будет, я написал рабочий процесс в ветке devops, рабочий процесс нацелен на работу с веткой тест и должен реагировать на пуш в ветку тест. Итог - он банально отсутствует в разделе actions, файл .github/workflows/devops_workflow.yml есть, а actions нет

**Comment by RAGHB24** [comment]
But I wasn't able to run the workflow if I directly created it in a feature branch. The workflow should have existed in the main at least once in it's lifetime, for it work.

**Comment by jneira-stratio** [comment]
Same here but it has happened recently, my github actions (using https://github.com/tibdex/backport) worked fine in feature branches until December 4th, after that i had to create the workflow in master to make feature branches work.
